### ISSUE TITLE
Using a prop as an object key to call a function, incorrectly reports missing in props validation

### ISSUE THREAD

**Issue by Standard8** [yannickcr/eslint-plugin-react#132]
Testcase:

```javascript
var TestComp1 = React.createClass({
  propTypes: {
    size: React.PropTypes.string.isRequired
  },

  render: function() {
    var foo = {
      "abc": ["1", "2", "3"]
    };

    var icons = foo[this.props.size].map(function() {
      return null;
    });
    return (<div>{icons}</div>);
  }
});
```
Output:
```testComp.jsx
  11:37  error  'size[].map' is missing in props validation  react/prop-types

✖ 1 problem (1 error, 0 warnings)
```

this.props.size is being used as the key for the object, not as an object itself.

**Comment by Standard8** [comment]
Forgot to say, this is a regression in 2.6.0.

**Comment by yannickcr** [comment]
It should come from the new nested props support. I will investigate.
